src/components/LoginRegister/LoginInput.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import { useApolloClient } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import { Input, Link, Button, Heading, Flex, VStack } from '@chakra-ui/react';
+import { login } from '../../data/actions/auth';
+
+const LoginInput = () => {
+  const client = useApolloClient();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [submitState, setSubmitState] = useState('');
+  const dispatch = useDispatch();
+
+  const handleClick = async e => {
+    e.preventDefault();
+    dispatch(login(email, password, client)).then(
+      loggedIn => {
+        if (!loggedIn) setSubmitState('FAIL');
+      },
+      error => {
+        setSubmitState('FAIL');
```

Comment:
just curioussss do you know if this is some "ESNext" shorthand for `.then` and `.catch`? I see this pattern throughout our codebase but I don't think I've ever seen it before anywhere else / I can't find the docs for this

Just curious if you know or you just copied the pattern 🤷

Reply:
Yeah, it's syntax for handling a Promise (read about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#chained_promises)). The first argument is a callback for the resolved case, and the second is for the rejected case. I think the alternative is to chain `.then` and `.catch`, but they should do the same thing!

Reply:
that is pre cool i feel like i am so behind on JS syntax there is new syntax every year lolol